consensus/istanbul/backend/handler.go

```diff
@@ -118,12 +118,19 @@ func (sb *Backend) HandleMsg(addr common.Address, msg p2p.Msg, peer consensus.Pe
 				return true, nil
 			}
 
-			fwdMsg := new(istanbul.Message)
-			if err := fwdMsg.FromPayload(data, nil); err != nil {
+			msg := new(istanbul.Message)
+			if err := msg.FromPayload(data, nil); err != nil {
 				sb.logger.Error("Failed to decode message from payload", "err", err)
 				return true, err
 			}
 
+			var fwdMsg *istanbul.ForwardMessage
+			err := msg.Decode(&fwdMsg)
```

Comment:
IIUC, we're not checking the signature on the message to forward because we've already verified that the fwd message came from a proxied peer, correct?

Reply:
Yes, that's correct.  I'll add a comment for that.